output/typescript/types.ts

```diff
@@ -3818,26 +3818,26 @@ export interface MappingTypesComplexNestedNestedProperty extends MappingTypesCor
   properties?: Record<PropertyName, MappingTypesProperty>
   include_in_parent?: boolean
   include_in_root?: boolean
-  type: 'nested'
+  type?: 'nested'
 }
 
 export interface MappingTypesComplexObjectObjectProperty extends MappingTypesCorePropertyBase {
   dynamic?: boolean | MappingDynamicMapping
   enabled?: boolean
   properties?: Record<PropertyName, MappingTypesProperty>
-  type: 'object'
+  type?: 'object'
```

Comment:
As discussed with @delvedor, we only need this single line. This is because field mapping uses `object` as the default type (see [bottom of this page](https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html)). In all other cases (other variants and containers that don't have a default type) `type` should be require.